rllib/algorithms/ppo/tf/ppo_tf_rl_module.py

```diff
@@ -24,85 +32,128 @@ class PPOTfRLModule(TfRLModule):
+        self.config.encoder_config.input_dim = self.config.observation_space.shape[0]
+        self.config.pi_config.input_dim = (
+            self.config.encoder_config.base_encoder_config.output_dim
         )
-
-        self.vf = TfMLP(
-            input_dim=self.config.encoder_config.output_dim,
-            output_dim=1,
-            hidden_layer_dims=self.config.vf_config.hidden_layer_dims,
-            hidden_layer_activation=self.config.vf_config.hidden_layer_activation,
+        self.config.vf_config.input_dim = (
+            self.config.encoder_config.base_encoder_config.output_dim
         )
 
+        if isinstance(self.config.action_space, gym.spaces.Discrete):
+            self.config.pi_config.output_dim = self.config.action_space.n
+        else:
+            self.config.pi_config.output_dim = self.config.action_space.shape[0] * 2
+        self.config.vf_config.output_dim = 1
+
+        # TODO(Artur): Unify to tf and torch setup with Catalog
+        self.encoder = self.config.encoder_config.build(framework="tf")
+        self.pi = self.config.pi_config.build(framework="tf")
+        self.vf = self.config.vf_config.build(framework="tf")
+
         self._is_discrete = isinstance(
             convert_old_gym_space_to_gymnasium_space(self.config.action_space),
             gym.spaces.Discrete,
         )
 
-    @override(TfRLModule)
-    def input_specs_train(self) -> List[str]:
-        return [SampleBatch.OBS, SampleBatch.ACTIONS]
-
-    @override(TfRLModule)
-    def output_specs_train(self) -> List[str]:
-        return [
-            SampleBatch.ACTION_DIST,
-            SampleBatch.VF_PREDS,
-        ]
+    @override(RLModule)
+    def input_specs_train(self) -> SpecDict:
+        if self._is_discrete:
+            action_spec = TFTensorSpecs("b")
+        else:
+            action_dim = self.config.action_space.shape[0]
+            action_spec = TFTensorSpecs("b, h", h=action_dim)
+
+        # TODO (Artur): Infer from encoder specs as soon as Policy supports RNN
+        spec_dict = self.input_specs_exploration()
+
+        spec_dict.update({SampleBatch.ACTIONS: action_spec})
+        if SampleBatch.OBS in spec_dict:
+            spec_dict[SampleBatch.NEXT_OBS] = spec_dict[SampleBatch.OBS]
+        spec = SpecDict(spec_dict)
+        return spec
+
+    @override(RLModule)
+    def output_specs_train(self) -> SpecDict:
+        spec = SpecDict(
+            {
+                SampleBatch.ACTION_DIST: Categorical
+                if self._is_discrete
+                else DiagGaussian,
+                SampleBatch.ACTION_LOGP: TFTensorSpecs("b", dtype=tf.float32),
+                SampleBatch.VF_PREDS: TFTensorSpecs("b", dtype=tf.float32),
+                "entropy": TFTensorSpecs("b", dtype=tf.float32),
+            }
+        )
+        return spec
 
-    @override(TfRLModule)
-    def _forward_train(self, batch: NestedDict):
+    @override(RLModule)
+    def _forward_train(self, batch: NestedDict) -> Mapping[str, Any]:
```

Comment:
The changes to the forward functions are not algorithmic, but separate encoder, policy- and value-head code a little nicer. I had to rewrite this anyway because the model interfaces changed.